libs/langchain/langchain/llms/bedrock.py

```diff
@@ -156,6 +172,26 @@ def _llm_type(self) -> str:
         """Return type of llm."""
         return "amazon_bedrock"
 
+    def _stream(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> Iterator[GenerationChunk]:
+        _model_kwargs = self.model_kwargs or {}
+        provider = self.model_id.split(".")[0]
+        params = {**_model_kwargs, **kwargs}
+        input_body = LLMInputOutputAdapter.prepare_input(provider, prompt, params)
+        body = json.dumps(input_body)
+        response = self.client.invoke_model_with_response_stream(
+            body=body,
+            modelId=self.model_id,
+            accept="application/json",
+            contentType="application/json",
+        )
+        return LLMInputOutputAdapter.prepare_output_stream(provider, response)
```

Comment:
Looks like we're not enforcing the stop tokens in `_stream`, unlike in `_call`, see line 234. What we probably want to do is something like
```py
buffer = ''
for event in stream:
  text = ...
  buffer += text
  if enforce_stop_tokens(buffer, stop).endswith(text):
    yield GenerationChunk(text=text)
  else:
    break